### PR TITLE
Line breaks in HtmlArea's macro TextArea fields are being removed #4083

### PR DESCRIPTION
**Pull request by ashklianko**
-using cke element's innerText property to get/set macro text to preserve line breaks

### CODE REVIEW

**Comment by sigdestad** [comment]
Has security aspect related to this been considered? @rymsha

**Comment by rymsha** [comment]
Yes, I would double check... 
`innerText` seem to not only allow line breaks, but allows basically anything, including `<script>` tags.
But I can't tell right now how bad it is.

**Comment by ashklianko** [comment]
CKEditor has built-in content filter (which we, ofcouse, can't rely on, and developers mention that) that strips and escapes vulnerable code (tags, attributes, etc. until they are allowed in the config). Basic XSS stuff definately won't  work because of innerText in that case. We also extract data from editor not by using such a dangerous stuff but by cke's provided api. The rest filtering is supposed to be on the backend.

**Comment by rymsha** [comment]
It is not only stored XSS. Other XSS flavors exist.
Depending on where and how the value fetched by innerText is inserted it may cause DOM based XSS.

I'm not familiar with getText() method. I would check what it does inside

**Comment by ashklianko** [comment]
If we talk about any types of the xss then the change in this task won't create additional risks, since cke content might be changed in multiple places, especially in Source dialog. Seems like cke vulnerability is a separate topic

**Comment by sigdestad** [comment]
Would be nice to know how this fix actually works though. Are we sure the "old way of working" will not break? Must be testes carefully with both XP and Guillotine rendering

**Comment by alansemenov** [comment]
@sigdestad this fix is not for HtmlArea

**Comment by sigdestad** [comment]
? But macros are used in Htmlarea

File: modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/MacroModalDialog.ts [hunk 1]
```diff
@@ -202,8 +202,8 @@ export class MacroModalDialog
     }
 
     private insertUpdatedMacroIntoTextArea(macroString: string) {
-        this.selectedMacro.element.setText(
-            this.selectedMacro.element.getText().replace(this.selectedMacro.macroText, macroString));
+        this.selectedMacro.element.$.innerText =
+            this.selectedMacro.element.$.innerText.replace(this.selectedMacro.macroText, macroString);
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/MacroModalDialog.ts:206]: @ashklianko Is it `setText` or `getText` that does sanitisation in CKE? Can we keep `setText` and only replace `getText` with `$.innerText`?

Comment (ashklianko) [reply modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/MacroModalDialog.ts:206]: setText and getText don't preserve line breaks, so just innerText works

**Comment by alansemenov** [comment]
Will be fixed in https://github.com/enonic/app-contentstudio/pull/4109

**Pull request closed without merging**